src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ManifestMetadataTableNode.cs

```diff
@@ -241,5 +250,24 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 alignment: 1,
                 definedSymbols: new ISymbolDefinitionNode[] { this });
         }
+
+        private const int GuidByteSize = 16;
+
+        public int ManifestAssemblyMvidTableSize => GuidByteSize * _manifestAssemblyMvids.Count;
+
+        internal byte[] GetManifestAssemblyMvidTableData()
+        {
+            byte[] manifestAssemblyMvidTable = new byte[ManifestAssemblyMvidTableSize];
+            for (int i = 0; i < _manifestAssemblyMvids.Count; i++)
+            {
+                Array.Copy(
+                    sourceArray: _manifestAssemblyMvids[i].ToByteArray(),
+                    sourceIndex: 0,
+                    destinationArray: manifestAssemblyMvidTable,
+                    destinationIndex: GuidByteSize * i,
+                    length: GuidByteSize);
```

Comment:
Nit: `Guid.TryWriteBytes(Span<byte> destination)` is more efficient and would avoid creating helper arrays.

Reply:
Good point, thanks for the suggestion, fixed in 4th commit.